test/e2e/expirable-airdrop.spec.ts

```diff
@@ -0,0 +1,302 @@
+  let root: any;
+  let tree: any;
+  let token: any;
+  let expirableAirdrop: any;
+  let airdropAmount: number = 10;
+  let airdropAmountBN: BigNumber = toUnit(airdropAmount);
+  let snapshot: SnapshotRestorer;
+  let alice: SignerWithAddress;
+  let bob: SignerWithAddress;
+  let carl: SignerWithAddress;
+  let random: SignerWithAddress;
+  const randomAddress: string = '0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045';
+
+  async function prepareAirdrop() {
+    let toDeposit = toUnit(airdropAmount * (accounts.length + 1));
+    await token.mint(expirableAirdrop.address, toDeposit);
+  }
+
+  before(async () => {
+    nowTimestamp = await time.latest();
+    expirationTimestamp = nowTimestamp + 3 * oneMonth; // 3 months
+    [deployer, ...accounts] = await ethers.getSigners();
+
+    [root, tree] = createMerkleRoot(accounts, airdropAmountBN);
+
+    // deploy erc20 mock
+    const TokenFactory = await ethers.getContractFactory('ERC20Mock');
+    token = await TokenFactory.deploy('Token', 'TKN');
+
+    // deploy contract
+    const ExpirableAirdropFactory = await ethers.getContractFactory('ExpirableAirdrop');
+    expirableAirdrop = await ExpirableAirdropFactory.deploy(deployer.address, token.address, expirationTimestamp, root);
+
+    // load accounts
+    alice = accounts[1];
+    bob = accounts[2];
+    carl = accounts[3];
+    await impersonateAccount(randomAddress);
+    await setBalance(randomAddress, toUnit(10000));
+    random = await ethers.getSigner(randomAddress);
+
+    // save snapshot for later
+    snapshot = await takeSnapshot();
+  });
+
+  beforeEach(async () => {
+    await snapshot.restore();
+  });
+
+  when('depositTokens()', () => {
+    let toDeposit: BigNumber;
+    let tx: Transaction;
+
+    given(async () => {
+      toDeposit = toUnit(airdropAmount * (accounts.length + 1));
+
+      // transfer some tokens
+      await token.mint(deployer.address, toDeposit);
+
+      // approve
+      await token.approve(expirableAirdrop.address, ethers.constants.MaxUint256);
+
+      // deposit
+      await expirableAirdrop.depositTokens(toDeposit);
+    });
+
+    then('airdrop is deposited', async () => {
+      // expect
+      expect(await token.balanceOf(expirableAirdrop.address)).equal(toDeposit);
+    });
+
+    then('event is emitted', async () => {
+      // expect
+      expect(tx).to.have.emit(expirableAirdrop, 'Deposited').withArgs(toDeposit);
+    });
+  });
+
+  when('claimAndSendToClaimee()', () => {
+    let leaf: any;
+    let proof: any;
+
+    given(async () => {
+      await prepareAirdrop();
+
+      leaf = getLeaf(alice.address, airdropAmountBN);
+      proof = tree.getHexProof(leaf);
+    });
+
+    when('airdrop is claimed by claimee', () => {
+      let tx: Transaction;
+
+      given(async () => {
+        tx = await expirableAirdrop.connect(alice).claimAndSendToClaimee(alice.address, airdropAmountBN, proof);
+      });
+
+      then('tokens are transferred to claimee', async () => {
+        // expect
```

Comment:
```suggestion

```
This will apply to all //expect comment, unless there is something I am missing 😋